src/daft-parquet/src/file.rs

```diff
@@ -94,33 +95,30 @@ impl ParquetReaderBuilder {
 
         let metadata = read_parquet_metadata(uri, size, io_client).await?;
         let num_rows = metadata.num_rows;
-        let schema =
```

Comment:
NOTE: schema inference is now deferred to only when `.build()` is called

This allows the builder to make better decisions about *whether* it needs to infer the schema (i.e. did the user provide one?) and, if so, *how* it should infer the schema (what flags did the user provide)?